elasticsearch/src/http/transport.rs

```diff
@@ -595,16 +701,134 @@ impl CloudConnectionPool {
 
 impl ConnectionPool for CloudConnectionPool {
     /// Gets a reference to the next [Connection]
-    fn next(&self) -> &Connection {
-        &self.connection
+    fn next(&self) -> Connection {
+        self.connection.clone()
+    }
+}
+
+/// A Connection Pool that manages a static connection of nodes
+#[derive(Debug, Clone)]
+pub struct MultiNodeConnectionPool<LoadBalancing = RoundRobin> {
+    inner: Arc<RwLock<MultiNodeConnectionPoolInner>>,
+    reseed_frequency: Option<Duration>,
+    load_balancing_strategy: LoadBalancing,
+    reseeding: Arc<AtomicBool>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MultiNodeConnectionPoolInner {
+    last_update: Option<Instant>,
+    connections: Vec<Connection>,
+}
+
+impl<LoadBalancing> ConnectionPool for MultiNodeConnectionPool<LoadBalancing>
+where
+    LoadBalancing: LoadBalancingStrategy + Clone,
+{
+    fn next(&self) -> Connection {
+        let inner = self.inner.read().expect("lock poisoned");
+        self.load_balancing_strategy
+            .try_next(&inner.connections)
+            .unwrap()
+    }
+
+    fn reseedable(&self) -> bool {
+        let inner = self.inner.read().expect("lock poisoned");
+        let reseed_frequency = match self.reseed_frequency {
+            Some(wait) => wait,
+            None => return false,
+        };
+        let last_update_is_stale = inner
+            .last_update
+            .as_ref()
+            .map(|last_update| last_update.elapsed() > reseed_frequency);
+        let reseedable = last_update_is_stale.unwrap_or(true);
+
+        return if !reseedable {
+            false
+        } else {
+            // Check if refreshing is false if so, sets to true atomically and returns old value (false) meaning refreshable is true
+            // If refreshing is set to true, do nothing and return true, meaning refreshable is false
+            !self
+                .reseeding
+                .compare_and_swap(false, true, Ordering::Relaxed)
```

Comment:
Call out on latest commit was introducing an `AtomicBool` so that multiple threads won't kick off multiple reseeds. Lmk if this implementation is sound. This should in theory only allow one thread to get `Reseedable() == true` in a given time due to atomic check and swap.

Reply:
This looks reasonable to me